modules/openapi-generator/src/main/resources/r/modelOneOf.mustache

```diff
@@ -139,25 +139,35 @@
 
     #' @description
     #' Serialize {{{classname}}} to JSON string.
-    #'
+    #' 
+    #' @param ... Parameters passed to `jsonlite::toJSON`
     #' @return JSON string representation of the {{{classname}}}.
-    toJSONString = function() {
+    toJSONString = function(...) {
+      simple <- self$toSimpleType()
       if (!is.null(self$actual_instance)) {
-        as.character(jsonlite::minify(self$actual_instance$toJSONString()))
+        json <- jsonlite::toJSON(simple, auto_unbox = TRUE, ...)
+        return(as.character(jsonlite::minify(json)))
       } else {
-        NULL
+        return(NULL)
       }
     },
 
     #' @description
-    #' Serialize {{{classname}}} to JSON.
-    #'
-    #' @return JSON representation of the {{{classname}}}.
+    #' Convert to an R object. This method is deprecated. Use `toSimpleType()` instead.
     toJSON = function() {
+      .Deprecated(new = "toSimpleType", msg = "Use the '$toSimpleType()' method instead since that is more learly named. Use '$toJSONString()' to get a JSON string")
+      return(self$toSimpleType())
+    },
+
+    #' @description
+    #' Convert {{{classname}}} to a base R type
+    #'
+    #' @return A base R type, e.g. a list or numeric/character array.
+    toSimpleType = function() {
       if (!is.null(self$actual_instance)) {
-        self$actual_instance$toJSON()
+        return(self$actual_instance$toSimpleType())
       } else {
-        NULL
+        return(NULL)
       }
```

Comment:
OK, I refactored this PR a bit since a realized that not all model types necessarily get transformed to R list types enroute to JSON.

The gist is that
1. I depreciated `toJSON` in favor of `toSimpleType` and made the behavior consistent (i.e., it returns an R list, or character vector, etc.). Previously `toJSON` might return a list or a JSON string, depending on the model type.
2. I added `toSimpleType` to all models
3. I modified `toJSONString` to use `toSimpleType()` to get the R object and then invoke the `jsonlite` transformations on that object (`toJSON` and `minifiy`).